mantidproject/mantid
# Add YAPF configuration and developer docs

Please merge mantidproject/thirdparty-msvc2015#45 when this is merged. 

**Description of work.**

Adds a basic YAPF configuration file (copied from https://github.com/DiamondLightSource/hebi) that produces reasonable results.

Adds developer documentation that describes how YAPF can be run to format changed Python files pre commit.

**To test:**

- RTFM
- Try out some of the commands in the docs
- See that they produce nicely formatted Python

*This does not require release notes* because it only affects developers.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
